### DISCUSSION TITLE
pushImage with transparency

### DISCUSSION THREAD

**Discussion by bobwolff68** [Bodmer/TFT_eSPI#1160] [Q&A - Sprites] [answered]
Hey there,
I was scratching my head as to why the compiler was erroring out on my code. I was staring at the TFT prototype:

`void TFT_eSPI::pushImage(int32_t x, int32_t y, int32_t w, int32_t h, const uint16_t *data, uint16_t transp)`

And then I realized I was trying to use a Sprite object and came to learn that the pushImage() function isn't supported inside of Sprites WITH transparency (and with PROGMEM/const data pref).

Is there a reason for this missing item other than it just hasn't been implemented?

Or is there some other obvious way that I can push a 16-bit color image into a Sprite area where there is a pixel value that should be transparent?

--robobob

**Comment by Bodmer** [comment, marked as answer]
Yes, not all possible permutations have been implemented.

You can push a sprite to another sprite with transparency.

**Reply by bobwolff68** [reply]
Ok cool - if I get back to that section of the design soon, I'll write up a Sprite version of pushImage with transparency option. I just wanted to make sure there weren't any particular reasons it doesn't exist aside from "work" ... :)

**Reply by renaudbarbier** [reply]
@Bodmer could you please share an example on how to push a sprite to an other sprite with transparency? I saw the function:
bool TFT_eSprite::pushToSprite(TFT_eSprite *dspr, int32_t x, int32_t y, uint16_t transp)

but did not managed to make it work. it seems the way I declare the destination sprite is not ok. (I just typed the sprite name in there)

**Reply by renaudbarbier** [reply]
by the way, pushing a cropped image (pushimage) with transparency to the tft would be great feature...

**Reply by Bodmer** [reply]
Pushing a cropped image can be achieved by using the TFT[ setViewport function](https://github.com/Bodmer/TFT_eSPI/blob/master/TFT_eSPI.h#L477-L478).

**Reply by renaudbarbier** [reply]
I meant cropped with TRANSPARENCY.

**Reply by renaudbarbier** [reply]
> @Bodmer could you please share an example on how to push a sprite to an other sprite with transparency? I saw the function: bool TFT_eSprite::pushToSprite(TFT_eSprite *dspr, int32_t x, int32_t y, uint16_t transp)
> 
> but did not managed to make it work. it seems the way I declare the destination sprite is not ok. (I just typed the sprite name in there)

what about this?

**Reply by Bodmer** [reply]
Example:
spr.pushToSprite(&dspr, x, y, TFT_BLACK);
